src/neo/SmartContract/NefFile.cs

```diff
@@ -110,20 +111,17 @@ public void Deserialize(BinaryReader reader)
         /// <returns>Return checksum</returns>
         public static uint ComputeChecksum(NefFile file)
         {
-            using (var ms = new MemoryStream())
-            using (var wr = new BinaryWriter(ms))
-            {
-                file.Serialize(wr);
-                wr.Flush();
+            using var ms = new MemoryStream();
+            using var wr = new BinaryWriter(ms, Encoding.UTF8, false);
 
-                // Read header without CRC
+            file.Serialize(wr);
+            wr.Flush();
 
-                Span<byte> buffer = stackalloc byte[HeaderSize - sizeof(uint)];
-                ms.Seek(0, SeekOrigin.Begin);
-                ms.Read(buffer);
+            // Read header without CRC
 
-                return BitConverter.ToUInt32(buffer.Sha256(), 0);
-            }
+            ms.Seek(0, SeekOrigin.Begin);
```

Comment:
I think we should still read the header.

Reply:
Fixed

Reply:
I think that the checksum should include the compiler version, what do you think?